Mathlib/Combinatorics/Quiver/Path/Vertices.lean

```diff
@@ -0,0 +1,262 @@
+/-
+Copyright (c) 2025 Matteo Cipollina. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Matteo Cipollina
+-/
+
+import Mathlib.Algebra.Order.Group.Nat
+import Mathlib.Algebra.Order.Sub.Basic
+import Mathlib.Combinatorics.Quiver.Path
+import Mathlib.Data.Finset.Dedup
+
+/-!
+# Path Vertices
+
+This file provides lemmas for reasoning about the vertices of a path.
+-/
+
+namespace Quiver.Path
+
+open List
+
+variable {V : Type*} [Quiver V]
+
+/-- The end vertex of a path. A path `p : Path a b` has `p.end = b`. -/
+def «end» {a : V} : ∀ {b : V}, Path a b → V
+  | b, _ => b
+
+@[simp]
+lemma end_cons {a b c : V} (p : Path a b) (e : b ⟶ c) : «end» (p.cons e) = c := rfl
+
+/-- The list of vertices in a path, including the start and end vertices. -/
+def vertices {a : V} : ∀ {b : V}, Path a b → List V
+  | _, nil => [a]
+  | _, cons p e => (p.vertices).concat («end» (p.cons e))
+
+/-- The set of vertices in a path. -/
+def vertices_set {a b : V} (p : Path a b) : Set V :=
+  {v | v ∈ p.vertices}
+
+/-- The finset of vertices in a path. -/
+def vertices_finset [DecidableEq V] {a b : V} (p : Path a b) : Finset V :=
+  p.vertices.toFinset
+
+/-- The finset of vertices in a path, excluding the final vertex. -/
+def init_finset [DecidableEq V] {a b : V} (p : Path a b) : Finset V :=
+  p.vertices.dropLast.toFinset
```

Comment:
I'm also a bit dubious about this definition and if we keep it we need a better name.